stdlib/src/mmap.rs

```diff
@@ -0,0 +1,798 @@
+            libc::MADV_DONTNEED => Advice::DontNeed,
+            #[cfg(target_os = "linux")]
+            libc::MADV_FREE => Advice::Free,
+            #[cfg(target_os = "linux")]
+            libc::MADV_DONTFORK => Advice::DontFork,
+            #[cfg(target_os = "linux")]
+            libc::MADV_DOFORK => Advice::DoFork,
+            #[cfg(target_os = "linux")]
+            libc::MADV_MERGEABLE => Advice::Mergeable,
+            #[cfg(target_os = "linux")]
+            libc::MADV_UNMERGEABLE => Advice::Unmergeable,
+            #[cfg(target_os = "linux")]
+            libc::MADV_HUGEPAGE => Advice::HugePage,
+            #[cfg(target_os = "linux")]
+            libc::MADV_NOHUGEPAGE => Advice::NoHugePage,
+            #[cfg(target_os = "linux")]
+            libc::MADV_REMOVE => Advice::Remove,
+            #[cfg(target_os = "linux")]
+            libc::MADV_DONTDUMP => Advice::DontDump,
+            #[cfg(target_os = "linux")]
+            libc::MADV_DODUMP => Advice::DoDump,
+            #[cfg(target_os = "linux")]
+            libc::MADV_HWPOISON => Advice::HwPoison,
+            _ => return Err(vm.new_value_error("Not a valid Advice value".to_owned())),
+        })
+    }
+
+    #[repr(C)]
+    #[derive(PartialEq, Eq, Debug)]
+    enum AccessMode {
+        Default = 0,
+        Read = 1,
+        Write = 2,
+        Copy = 3,
+    }
+
+    impl TryFromBorrowedObject for AccessMode {
+        fn try_from_borrowed_object(vm: &VirtualMachine, obj: &PyObject) -> PyResult<Self> {
+            let i = u32::try_from_borrowed_object(vm, obj)?;
+            Ok(match i {
+                0 => Self::Default,
+                1 => Self::Read,
+                2 => Self::Write,
+                3 => Self::Copy,
+                _ => return Err(vm.new_value_error("Not a valid AccessMode value".to_owned())),
+            })
+        }
+    }
+
+    #[pyattr]
+    use libc::{
+        MADV_DONTNEED, MADV_FREE_REUSABLE, MADV_FREE_REUSE, MADV_NORMAL, MADV_RANDOM,
+        MADV_SEQUENTIAL, MADV_WILLNEED, MAP_ANON, MAP_ANONYMOUS, MAP_PRIVATE, MAP_SHARED,
+        PROT_READ, PROT_WRITE,
+    };
+
+    #[cfg(target_os = "linux")]
+    use libc::{
+        MADV_DODUMP, MADV_DOFORK, MADV_DONTDUMP, MADV_DONTFORK, MADV_FREE, MADV_HUGEPAGE,
+        MADV_HWPOISON, MADV_MERGEABLE, MADV_NOHUGEPAGE, MADV_REMOVE, MADV_SOFT_OFFLINE,
+        MADV_UNMERGEABLE,
+    };
+
+    #[cfg(all(target_os = "linux", target_arch = "x86_64", target_env = "gnu"))]
+    use libc::{MAP_DENYWRITE, MAP_EXECUTABLE, MAP_POPULATE};
+
+    #[pyattr]
+    const ACCESS_DEFAULT: u32 = AccessMode::Default as u32;
+    #[pyattr]
+    const ACCESS_READ: u32 = AccessMode::Read as u32;
+    #[pyattr]
+    const ACCESS_WRITE: u32 = AccessMode::Write as u32;
+    #[pyattr]
+    const ACCESS_COPY: u32 = AccessMode::Copy as u32;
+
+    #[derive(Debug, Clone)]
+    enum MmapObj {
+        //TODO: better container to keep MmapMut
+        Write(Arc<PyMutex<MmapMut>>),
+        Read(Arc<Mmap>),
```

Comment:
```suggestion
        Write(PyRc<PyMutex<MmapMut>>),
        Read(PyRc<Mmap>),
```

`PyRc` abstracts over `Arc` and `Rc`, depending on if a platform supports threading.